ArduCopter/RC_Channel.cpp

```diff
@@ -48,6 +48,16 @@ bool RC_Channels_Copter::has_valid_input() const
     return true;
 }
 
+// allow disabling AP_Arming throttle check
+bool RC_Channels_Copter::arming_check_throttle() const {
+    if ((copter.g.throttle_behavior & THR_BEHAVE_FEEDBACK_FROM_MID_STICK) != 0) {
```

Comment:
I wonder if we should make a virtual method which returns this information instead of the approach taken here.

That would allow Plane to provide the same information to allow sprung-throttle arming.

Reply:
This only works because Copter is already doing throttle checks, so were not skipping all throttle checks. I suspect the real solution is to move copters throttle checks up and then we could do a virtual vehicle method as you suggest.